### ISSUE TITLE
In search results, make indicator code the link text

### ISSUE THREAD

**Issue by LucyGwilliamAdmin** [open-sdg/open-sdg#845]
**Page 113 in DAC report**

**Issue from DAC report:**
In search results (see below image), link text is long and non-descriptive as it does not describe the destination of the link, only the passage which contains the search text.

**Solution from DAC report:**
Please consider implementing the link destination as the link text, with the bulk of text which corresponds to the passage containing the search word implemented separately.

**Example from DAC report:**

```
<h3 class="title">Proportion of schools with access to (a) electricity; (b)
  the Internet for pedagogical purposes; (c) computers for pedagogical
  purposes; (d) adapted infrastructure and materials for students with
  disabilities; (e) basic drinking water; (f) single-sex basic sanitation
  facilities; and (g) basic handwashing facilities (as per the WASH indicator
  definitions)</h3>
<a href="/4-a-1">
    <span class="url">Indicator 4.a.1</span>
</a>
```

instead of

```
<a href="/4-a-1">
    <h3 class="title">Proportion of schools with access to (a) electricity; (b)
      the Internet for pedagogical purposes; (c) computers for pedagogical
      purposes; (d) adapted infrastructure and materials for students with
      disabilities; (e) basic drinking water; (f) single-sex basic sanitation
      facilities; and (g) basic handwashing facilities (as per the WASH indicator
      definitions)</h3>
    <span class="url">Indicators » /4-a-1</span>
</a>
```

![image](https://user-images.githubusercontent.com/43404358/91590959-11993780-e954-11ea-9e12-6e1038ba65bf.png)

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin @AnnCorp @phillipgardiner A correction here: the issue description says this:

> In search results (see below image), link text is long and non-descriptive as it does not describe the destination of the link, only the passage which contains the search text.

But the link text is not just "the passage which contains the search text". It's actually the title of the indicator/page/etc.. I'm not sure if this makes a difference to this issue, but I thought it was worth mentioning. Admittedly some indicator titles can get very long.

I think this needs some consideration before implementing the suggestion, because it also has an effect on how sighted users experience the search results. Whatever becomes the link is what sighted users will click on, and we should make sure it is still visually intuitive.

My initial reaction is that this accessibility suggestion should go hand-in-hand with some feedback from Phil. The DAC suggestion can be boiled down to: "In the search results, some indicator titles can be so long that it is frustrating for users with screen-readers. Link text should be the minimum length possible to convey the necessary information."

The DAC suggestion represents the opinion that the indicator number is more important than the indicator title, in search results. In other words, according to the DAC suggestion it's more important that the screen-reader read out "Indicator 1.1.1" than "Proportion of population below the international poverty line, by sex, age, employment status and geographical location (urban/rural)".

I think that's up for discussion here - I'm fine either way. Also, in theory, they could be combined (eg "Indicator 1.1.1: Proportion of population below the..."), although that makes the link text even longer. Having the "Indicator 1.1.1" at the beginning ensures that screen-readers announce that first, and users can tab to the next link if they don't want to sit through the rest of a long title.

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin @phillipgardiner @jwestw @AnnCorp I know we're waiting on feedback from DAC, but in the meantime, one possible user-story that would support adding the indicator number might be this:

> I'm a researcher trying to find all the indicators that have to do with poverty. So I search for "Poverty". In the search results, it would be helpful to see/hear the indicator number for each indicator in the results, so that I don't have to click on each result to find the number.

That could be satisfied by adding "Indicator [x]: " in front of the title that already displays. Presumably we would also add "Goal [x]: " in front of the titles of any goals in the search results.

**Comment by phillipgardiner** [comment]
@brockfanning do mean like what we used to have a while back? (see screenshot below)

![image](https://user-images.githubusercontent.com/60919820/93312026-9e6e3d00-f7fe-11ea-93b1-b52047377248.png)

**Comment by brockfanning** [comment]
We'll try adding back the number (only) for indicators, and for goals "Goal [x] ".

**Comment by LucyGwilliamAdmin** [comment]
Reply from DAC:

>It definitely assists to have the identifier at the start of the link text, and the link purpose/destination is clearer. Again, I would say that the link text is non-descriptive as it describes the indicator more than it describes the link destination page. And the link text is still very long which may cause confusion. For example, some of these contain lists:

>Programmatically, where you say ‘1.1.1 Proportion of population below the international poverty line, by sex, age, employment status and geographical location (urban/rural)’ is split into headings, it isn’t from the perspective of screen reader users - the heading describing the purpose of the page is ‘Indicator 1.1.1’ so I would be inclined to say if that heading is descriptive enough, then the link text ‘Indicator 1.1.1’ should be sufficient, especially if the additional text is still there alongside it.

```
<div>
  <h1>Indicator 1.1.1</h1>
  <p class="lead-copy">Proportion of population below the international poverty line, by sex, age, employment status and geographical location (urban/rural)</p>
</div>
```

>But if this text is intended to be a part of the heading, we would suggest the following:

```
<hgroup>
  <h1>Indicator 1.1.1</h1>
  <h2 class="lead-copy">Proportion of population below the international poverty line, by sex, age, employment status and geographical location (urban/rural)</h2>
</hgroup>
```

> But in terms of addressing the A issue and ensuring that the link purpose is clear, I would say that prefixing the links with the destination page heading is probably sufficient to address this, and that the issue of it being overly long and non-descriptive is a usability issue.

**Comment by brockfanning** [comment]
We'll test out adding the indicator name to indicator page titles and H1.

**Comment by brockfanning** [comment]
Update: we'll add "Indicator" in front of the number, with a dash, and also a dash for the goal links.
